redis/src/aio/multiplexed_connection.rs

```diff
@@ -520,15 +580,180 @@ impl MultiplexedConnection {
+        if !missing_key_indexes.is_empty() {
+            request.write_all(&tail_buf).expect("TODO: panic message");
+            mget_cmd.write_packed_command(&mut request);
+            crate::cmd("EXEC").write_packed_command(&mut request);
+            cmd_count += 2;
+            cmd_count += missing_key_indexes.len();
+            let result = self
+                .pipeline
+                .send_recv(request, Some((0, cmd_count)), self.response_timeout)
+                .await
+                .map_err(|err| {
+                    err.unwrap_or_else(|| {
+                        RedisError::from(io::Error::from(io::ErrorKind::BrokenPipe))
+                    })
+                });
+            let responses = result?;
+            if let Value::Array(mut responses) = responses {
+                if let Value::Array(mut value_list_f) = responses.pop().unwrap() {
+                    if let Value::Array(mut value_list) = value_list_f.pop().unwrap() {
+                        for missing_key_index in missing_key_indexes.iter().rev() {
+                            let value = value_list.pop().unwrap();
+                            let pttl = value_list_f.pop().unwrap();
+                            let index = *missing_key_index;
+                            if let Some(Some(sender)) = senders.get(index) {
+                                let pttl: i64 = crate::FromRedisValue::from_redis_value(&pttl)?;
+                                let ci = cache_information_vec.get_mut(index).unwrap();
+                                if pttl >= 0 {
+                                    ci.client_side_ttl = Some(Duration::from_millis(pttl as u64));
+                                }
+                                self.cache_manager
+                                    .insert_with_guard(ci, sender, value.clone());
+                            } else {
+                                response[*missing_key_index] = value;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        Ok(Value::Array(response))
+    }
     /// Sends an already encoded (packed) command into the TCP socket and
     /// reads the single response from it.
     pub async fn send_packed_command(&mut self, cmd: &Cmd) -> RedisResult<Value> {
-        self.pipeline
-            .send_single(cmd.get_packed_command(), self.response_timeout)
+        #[cfg(feature = "cache-aio")]
+        let cache_information = match self.cache_manager.cache_config.mode {
+            CacheMode::All => cmd.compute_cache_information(),
+            CacheMode::OptIn => {
+                if cmd.has_opt_in_cache() {
+                    cmd.compute_cache_information()
+                } else {
+                    None
+                }
+            }
+            CacheMode::None => None,
+        };
+        #[cfg(feature = "cache-aio")]
+        let (cmd_bytes, cmd_count, notifier) = if let Some(ci) = &cache_information {
```

Comment:
there's a lot of code here and above which isn't multiplexer specific, and seems to also be relevant for other connections - anything which involves creating new commands, for example. Please move that to the cache module.

Reply:
I've moved some but not all. Is this enough ? I prefer not moving everything to cache module when potential overlap is unknown.